packages/zql/src/ivm/exists.ts

```diff
@@ -144,19 +144,27 @@ export class Exists implements Operator {
               size = this.#fetchSize(change.node);
             }
             if (size === 1) {
-              const type = this.#not ? 'remove' : 'add';
-              // The node for the remove pushed below will contain the child
-              // added by this change in its
-              // relationships[this.#relationshipName],
-              // so this child add needs to be sent first.  This balance
-              // is important for outputs doing ref counting,
-              if (type === 'remove') {
-                this.#output.push(change);
+              if (this.#not) {
+                // Since the add child change currently being processed is not
+                // pushed to output, the added child needs to be excluded from
+                // the remove being pushed to output (since the child has
+                // never been added to the output).
+                this.#output.push({
+                  type: 'remove',
+                  node: {
+                    row: change.node.row,
+                    relationships: {
+                      ...change.node.relationships,
+                      [this.#relationshipName]: () => [],
```

Comment:
I think it'd be clearer if the comment above `this.#output.push` was placed here where you're returning an empty relationship.